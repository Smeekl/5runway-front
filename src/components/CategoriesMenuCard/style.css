.categories-menu__img {
  border-radius: 10px;
  max-width: 200px;
  max-height: 178px;
}

.categories-menu__item {
  position: relative;
  border-radius: 20px;
}

.categories-menu__title {
  margin: 0;
  color: var(--color-secondary-white);
  width: 100px;
  top: 90px;
  left: 15px;
  position: absolute;
  font-weight: 700;
  font-size: 14px;
  line-height: 17px;
}
.categories-menu__item:hover {
  opacity: 0.8;
}
