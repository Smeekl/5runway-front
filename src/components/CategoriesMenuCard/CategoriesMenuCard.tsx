import './style.css';

interface ICategoriesMenuCardProps {
  name: string;
  imgUrl: string;
}

export const CategoriesMenuCard = ({
  name,
  imgUrl,
}: ICategoriesMenuCardProps) => (
  <div className="categories-menu__item" data-testid="wrapper">
    <img
      className="categories-menu__img"
      src={imgUrl}
      alt=""
      data-testid="image"
    />
    <p className="categories-menu__title" data-testid="title">
      {name}
    </p>
  </div>
);
