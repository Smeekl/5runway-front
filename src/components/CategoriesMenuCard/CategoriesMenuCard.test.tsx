import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { CategoriesMenuCard } from './CategoriesMenuCard';

describe('CategoriesMenuCard UI tests', () => {
  test('should render correctly', () => {
    const { container } = render(
      <MemoryRouter>
        <CategoriesMenuCard />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });

  test('should be in the document', () => {
    const { container } = render(
      <MemoryRouter>
        <CategoriesMenuCard />
      </MemoryRouter>
    );
    expect(container).toBeInTheDocument();
  });
});
