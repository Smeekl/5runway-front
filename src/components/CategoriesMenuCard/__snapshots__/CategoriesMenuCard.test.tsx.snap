// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Conditions UI tests should render correctly 1`] = `
<div>
  <div
    class="categories-menu__item"
    data-testid="wrapper"
  >
    <img
      alt=""
      class="categories-menu__img"
      data-testid="image"
    />
    <p
      class="categories-menu__title"
      data-testid="title"
    />
  </div>
</div>
`;
